nunjucks/src/lib.js

```diff
@@ -167,9 +167,58 @@ function isObject(obj) {
 
 exports.isObject = isObject;
 
+/**
+ * @param {string|number} attr
+ * @returns {(string|number)[]}
+ * @private
+ */
+function _prepareAttributeParts(attr) {
+  if (!attr) {
+    return [];
+  }
+
+  if (typeof attr === 'string') {
+    return attr.split('.');
+  }
+
+  return [attr];
+}
+
+/**
+ * @param {string}   attribute      Attribute value. Dots allowed.
+ * @returns {function(Object): *}
+ */
+function getAttrGetter(attribute) {
+  const parts = _prepareAttributeParts(attribute);
+
+  return function attrGetter(item) {
+    let _item = item;
+
+    function rollback() {
+      _item = undefined;
+    }
+
+    for (let i = 0; i < parts.length; i++) {
+      const part = parts[i];
+
+      // If item is not an object, and we still got parts to handle, it means
+      // that something goes wrong. Just roll out to undefined in that case.
+      if (hasOwnProp(_item, part)) {
+        _item = _item[part];
+      } else {
+        rollback();
```

Comment:
What's the advantage of the rollback function? Wouldn't it be functionally equivalent to simply `return;` here?

Reply:
You right, It was added in previous version, when there was default value in arguments. I will replace it with return for now.